packages/@aws-cdk/sdk-v2-to-v3-adapter/lib/coerce-api-parameters.ts

```diff
@@ -0,0 +1,54 @@
+import { UINT8ARRAY_PARAMETERS } from './parameter-types';
+
+type ApiParameters = { [param: string]: any };
+
+/**
+ * Given a minimal AWS SDKv3 call definition (service, action, parameters),
+ * coerces nested parameter values into a Uint8Array if that's what the SDKv3 expects.
+ */
+export function coerceApiParametersToUint8Array(service: string, action: string, parameters: ApiParameters = {}): ApiParameters {
+  const pathsToCoerce = UINT8ARRAY_PARAMETERS?.[service.toLowerCase()]?.[action.toLowerCase()] ?? [];
+  for (const path of pathsToCoerce) {
+    coerceToUint8Array(parameters, path.split('.'));
+  }
+  return parameters;
+}
+
+/**
+ * Given an object and a path, traverse to the leaf and coerce the value into a Uint8Array.
+ * This function mutates the original object.
```

Comment:
I think this implementation can be made simpler if we return the coerced values, but make no guarantee that we won't mutate the input objects.

So for primitives and arrays we return a fresh value, but we mutate objects in place (and then return it at the end).